File: Engineering Concepts/DSA-Java/financialforecasting/FinancialForecast.java
package com.cognizant.dsa.financialforecasting;

/*
 * Recursive Financial Forecasting
 *
 * Formula:
 * Future Value = Present Value × (1 + Growth Rate)^Years
 *
 * Time Complexity: O(n)
 * Space Complexity: O(n)
 *
 * The function calls itself once for each year
 * until the base condition (years == 0) is reached.
 *
 * Optimization:
 * An iterative approach or direct mathematical
 * formula can reduce recursive call overhead.
 */
public class FinancialForecast {

    public static double predictFutureValue(double currentValue,
                                            double growthRate,
                                            int years) {

        if (years == 0) {
            return currentValue;
        }

        return predictFutureValue(
                currentValue * (1 + growthRate),
                growthRate,
                years - 1
        );
    }
}

File: Engineering Concepts/DSA-CSharp/Exercise7_FinancialForecasting/FinancialForecast.cs
namespace Exercise7_FinancialForecasting
{
    public class FinancialForecast
    {

        /*
         * Recursive approach
         *
         * Time Complexity : O(n)
         * Space Complexity : O(n)
         */

        public static double PredictFutureValue(
            double currentValue,
            double growthRate,
            int years)
        {

            if (years == 0)
                return currentValue;


            return PredictFutureValue(

                currentValue * (1 + growthRate),

                growthRate,

                years - 1

                );
        }
    }
}